### PR TITLE
Initial check in to add firefly-hammer-x suite #12641

### PR DESCRIPTION
**Pull request by yuriw**
@liewegas, @athanatos, @tmuthamizhan  FYI

Incorporated Sage's comments, stress-split will have to be modified 

Signed-off-by: Yuri Weinstein <yweinste@redhat.com>

### CODE REVIEW

File: suites/upgrade/firefly-hammer-x/parallel/2-workload/rados_api.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+workload:
+   sequential:
+   - workunit:
+       branch: firefly
+       clients:
+         client.0:
+         # Note: check if the test scrip used is good
+         - rados/test-upgrade-v9.0.1.sh
```
Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/parallel/2-workload/rados_api.yaml]: this is a firefly -> hammer upgrade, so we should do whatever the firefly-x suite does here

File: suites/upgrade/firefly-hammer-x/parallel/3-upgrade-sequence/upgrade-all.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+upgrade-sequence:
+   sequential:
+   - install.upgrade:
+       mon.a:
+         branch: hammer
+       mon.b:
+         branch: hammer
+   - print: "**** done install.upgrade hammer for mon.a and mon.b"
+   - ceph.restart: [mon.a, mon.b, mon.c, mds.a, osd.0, osd.1, osd.2, osd.3]
```
Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/parallel/3-upgrade-sequence/upgrade-all.yaml]: i think this should space out the restarts by 60s each

File: suites/upgrade/firefly-hammer-x/parallel/5-workload/rados_api.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+workload2:
+  sequential:
+  - workunit:
+      branch: hammer
+      clients:
+        client.0:
+        - rados/test.sh
```
Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/parallel/5-workload/rados_api.yaml]: *here* is hwer eyou want to do test-v9.0.1.sh ... or whatever hammer-x does!

File: suites/upgrade/firefly-hammer-x/parallel/6-upgrade-sequence/upgrade-by-daemon.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+   - ceph.restart:
+       daemons: [mon.b, mon.c]
+       wait-for-healthy: true
+   - sleep:
+       duration: 60
+   - ceph.restart:
+       daemons: [osd.2, osd.3]
+       wait-for-healthy: true
+   - sleep:
+       duration: 60
```
Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/parallel/6-upgrade-sequence/upgrade-by-daemon.yaml]: this is what the firefly->hammer upgrade sequence should look like i think

Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/parallel/6-upgrade-sequence/upgrade-by-daemon.yaml]: (but look at what firefly-x does to see!)

File: suites/upgrade/firefly-hammer-x/stress-split/02-partial-upgrade-hammer/firsthalf.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+tasks:
+- install.upgrade:
+    osd.0:
+      branch: hammer
+- print: "**** done install.upgrade hammer"
+- ceph.restart:
+    daemons: [osd.0, osd.1, osd.2]
```
Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/stress-split/02-partial-upgrade-hammer/firsthalf.yaml]: restart 2 out of 3 mons too.. whichever ones are on that role

File: suites/upgrade/firefly-hammer-x/stress-split/06-monb-upgrade-hammer/monb.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+tasks:
+- ceph.restart:
+    daemons: [mon.b]
+    wait-for-healthy: false
+    wait-for-osds-up: true
+- print: "**** done restarted mon.b 06-monb-upgrade-hammer/monb.yaml"
```
Comment (liewegas) [inline suites/upgrade/firefly-hammer-x/stress-split/06-monb-upgrade-hammer/monb.yaml]: need to upgrade the other 3 osds, and the other mon, and mds here, right?  everything to hammer?

File: suites/upgrade/firefly-hammer-x/parallel/2-workload/rados_api.yaml (hunk 1 above)
Comment (yuriw) [inline suites/upgrade/firefly-hammer-x/parallel/2-workload/rados_api.yaml]: changed to "- rados.test.sh" 
it is then

File: suites/upgrade/firefly-hammer-x/parallel/3-upgrade-sequence/upgrade-all.yaml (hunk 2 above)
Comment (yuriw) [inline suites/upgrade/firefly-hammer-x/parallel/3-upgrade-sequence/upgrade-all.yaml]: done, pls take a look

File: suites/upgrade/firefly-hammer-x/parallel/5-workload/rados_api.yaml (hunk 3 above)
Comment (yuriw) [inline suites/upgrade/firefly-hammer-x/parallel/5-workload/rados_api.yaml]: modified to run "rados/test-upgrade-v9.0.1.sh"

**Comment by yuriw** [comment]
@liewegas - Incorporated your comments, pls review

File: suites/upgrade/firefly-hammer-x/parallel/3-upgrade-sequence/upgrade-all.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+upgrade-sequence:
```
Comment (yuriw) [inline suites/upgrade/firefly-hammer-x/parallel/3-upgrade-sequence/upgrade-all.yaml]: Sage, fixed this

File: suites/upgrade/firefly-hammer-x/parallel/7-final-workload/rados_mon_thrash.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+tasks:
+   - sequential:
+      - mon_thrash:
+          revive_delay: 20
+          thrash_delay: 1
+      - workunit:
+          clients:
+            client.1:
+            - rados/test-upgrade-v9.0.1.sh
```
Comment (yuriw) [inline suites/upgrade/firefly-hammer-x/parallel/7-final-workload/rados_mon_thrash.yaml]: Sage, I rearranged loads : 
in step 5 running "rados/test"
in step 7 rados/test-upgrade-v9.0.1.sh
This is a bit different what you said first time, so pls confirm.

**Pull request merged**